### ISSUE TITLE
Git repo not building

### ISSUE THREAD

**Issue by Hordeking** [OoliteProject/oolite#281]
I seem to be having problems building the code nowadays, where I didn't used to about a year ago. Not sure why this is a problem now. Seems to be caused by missing objc.make and common.make files that aren't coming in with the repo.

Command used:
make -f Makefile release

Seems to also fail with the same issue when I attempt to make 1.82 as well.

> make[4]: Entering directory '/home/jd/programming/oolite/deps/mozilla/js/src/build-debug/config'
> make[4]: Nothing to be done for 'tools'.
> make[4]: Leaving directory '/home/jd/programming/oolite/deps/mozilla/js/src/build-debug/config'
> make[4]: Entering directory '/home/jd/programming/oolite/deps/mozilla/js/src/build-debug/editline'
> make[4]: Nothing to be done for 'tools'.
> make[4]: Leaving directory '/home/jd/programming/oolite/deps/mozilla/js/src/build-debug/editline'
> make[4]: Entering directory '/home/jd/programming/oolite/deps/mozilla/js/src/build-debug/shell'
> make[4]: Nothing to be done for 'tools'.
> make[4]: Leaving directory '/home/jd/programming/oolite/deps/mozilla/js/src/build-debug/shell'
> make[3]: Leaving directory '/home/jd/programming/oolite/deps/mozilla/js/src/build-debug'
> if test -d ./dist/bin ; then touch ./dist/bin/.purgecaches ; fi
> make[2]: Leaving directory '/home/jd/programming/oolite/deps/mozilla/js/src/build-debug'
> touch deps/mozilla/js/src/build-debug/build_stamp
> make[1]: Leaving directory '/home/jd/programming/oolite'
> make -f GNUmakefile debug=yes strip=no
> make[1]: Entering directory '/home/jd/programming/oolite'
> GNUmakefile:1: /common.make: No such file or directory
> GNUmakefile:433: /objc.make: No such file or directory
> make[1]: *** No rule to make target '/objc.make'.  Stop.
> make[1]: Leaving directory '/home/jd/programming/oolite'
> Makefile:63: recipe for target 'debug' failed
> make: *** [debug] Error 2

**Comment by AnotherCommander** [comment]
Can you please provide some additional information about the system you are building one and the build environment? Looks like Linux from what I see, but there is not much more I can say from the provided data.

Is this from a recent repository checkout? I trust that the submodules were pulled in correctly using `git submodule update --init` before building, correct? Also, in case you forked the repository, make sure you have followed the instructions shown at the bottom of the Readme.md file regarding submodules.

**Comment by AnotherCommander** [comment]
> GNUmakefile:1: /common.make: No such file or directory
GNUmakefile:433: /objc.make: No such file or directory

The above messages seem to indicate that you may have a broken GNUstep installation or some environment variables setting the paths to GNUstep folders may not be present. common.make and objc.make should be available and accessible in a standard GNUstep install.  What is the version of GNUstep used to build? Is it possible that you may have changed version recently?

**Comment by Hordeking** [comment]
Running Linux Mint 18.3 (which is essentially Ubuntu 16.04 Xenial). This is a fresh installation. The build seemed to get quite a ways along before choking (I only posted the very end because that's all I could grab from the terminal without rebuilding).

Gnustep-core-devel 7.8 is installed, along with libsdl1.2.

This one is a headscratcher for me, as when I used to build the nightlies, I didn't have any trouble whatsoever when using the basic gnustep stuff. I did a whole lot of searching and couldn't even find anyone having problems with common.make and objc.make, no one responded on irc, so eventually I threw my hands up, posted about it here, and went to sleep.

**Comment by kanthoney** [comment]
Have you set the `GNUSTEP_MAKEFILES` environment variable? I've got the line

```
export GNUSTEP_MAKEFILES=/usr/lib64/GNUstep/Makefiles
```

in my `.bashrc`. Mint might have a different location, of course.

**Comment by Hordeking** [comment]
That got it. I also needed to install libopenal-dev, but that was fairly obvious once I set GNUSTEP_MAKEFILE. I appreciate it.

I do wonder if maybe we should include this this information in a summary at the top of Doc/README_LINUX.txt, along with a very specific list of required dependencies, like libopenal-dev and so on. Having a BUILD.txt file or something with the various full recipes to build on a few different architectures and their gotchas might be helpful, too.
